client-server-client/src/main/scala/zio/raft/client/RaftClient.scala

```diff
@@ -0,0 +1,461 @@
+              
+              case _ =>
+                ZIO.logWarning(s"Session rejected: $reason").as(Disconnected)
+            }
+          } else {
+            ZIO.logWarning("Nonce mismatch, ignoring SessionRejected").as(this)
+          }
+        
+        case StreamEvent.ServerMsg(other) =>
+          ZIO.logWarning(s"Unexpected message in ConnectingExistingSession: ${other.getClass.getSimpleName}").as(this)
+        
+        case StreamEvent.Action(ClientAction.SubmitCommand(_, promise)) =>
+          promise.fail(new IllegalStateException("Not connected")).ignore.as(this)
+        
+        case StreamEvent.Action(ClientAction.Disconnect) =>
+          ZIO.succeed(Disconnected)
+        
+        case _ =>
+          ZIO.succeed(this)
+      }
+    }
+    
+    def nextAddress: Option[(String, ConnectingExistingSession)] = {
+      val nextIndex = currentAddressIndex + 1
+      if (nextIndex < addresses.length) {
+        Some((addresses(nextIndex), copy(currentAddressIndex = nextIndex)))
+      } else {
+        None
+      }
+    }
+  }
+  
+  case class Connected(
+    sessionId: SessionId,
+    capabilities: Map[String, String],
+    createdAt: Instant,
+    serverRequestTracker: ServerRequestTracker,
+    nextRequestId: Ref[RequestId],
+    pendingRequests: Map[RequestId, Promise[Throwable, ByteVector]],
+    addresses: List[String]
+  ) extends ClientState {
+    override def stateName: String = s"Connected($sessionId)"
+    
+    override def handle(
+      event: StreamEvent,
+      transport: ZmqClientTransport,
+      config: ClientConfig,
+      serverRequestQueue: Queue[ServerRequest]
+    ): UIO[ClientState] = {
+      event match {
+        case StreamEvent.Action(ClientAction.Disconnect) =>
+          for {
+            _ <- transport.sendMessage(CloseSession(ClientShutdown)).orDie
+            _ <- ZIO.logInfo("Disconnected")
+          } yield Disconnected
+        
+        case StreamEvent.Action(ClientAction.SubmitCommand(payload, promise)) =>
+          for {
+            requestId <- nextRequestId.getAndUpdate(_.next)
+            now <- Clock.instant
+            request = ClientRequest(requestId, payload, now)
+            _ <- transport.sendMessage(request).orDie
+          } yield copy(pendingRequests = pendingRequests.updated(requestId, promise))
+        
+        case StreamEvent.ServerMsg(ClientResponse(requestId, result)) =>
+          pendingRequests.get(requestId) match {
+            case Some(promise) =>
+              promise.succeed(result).ignore.as(
+                copy(pendingRequests = pendingRequests.removed(requestId))
+              )
+            case None =>
+              ZIO.logWarning(s"Response for unknown request: $requestId").as(this)
+          }
+        
+        case StreamEvent.ServerMsg(KeepAliveResponse(_)) =>
+          ZIO.succeed(this)
+        
+        case StreamEvent.ServerMsg(serverRequest: ServerRequest) =>
+          for {
```

Comment:
you should check with the tracker and update it. If it not the next one, dropped it (and log. If it is the next one, add to queue, update the tracker and send the ack. If we already approved it, don't add it to the queue, but ack the message.